packages/cursorless-everywhere-talon-core/src/registerCommands.ts

```diff
@@ -0,0 +1,49 @@
+import { CURSORLESS_COMMAND_ID } from "@cursorless/common";
+import type { CommandApi } from "@cursorless/cursorless-engine";
+import type { TalonJsIDE } from "./ide/TalonJsIDE";
+import type { Talon } from "./types/talon.types";
+
+export function registerCommands(
+  talon: Talon,
+  ide: TalonJsIDE,
+  commandApi: CommandApi,
+) {
+  const ctx = talon.Context();
+  ctx.matches = "not app: vscode";
+
+  ctx.action_class("user", {
+    private_cursorless_run_rpc_command_no_wait(
+      commandId: string,
+      command: unknown,
+    ): void {
+      void runCommand(commandId, command);
+    },
+
+    async private_cursorless_run_rpc_command_get(
+      commandId: string,
+      command: unknown,
+    ): Promise<unknown> {
+      return await runCommand(commandId, command);
+    },
+  });
+
+  async function runCommand(
+    commandId: string,
+    command: unknown,
+  ): Promise<unknown> {
+    try {
+      if (commandId !== CURSORLESS_COMMAND_ID) {
+        throw Error(`Unknown command ID: ${commandId}`);
+      }
+
+      const editorState =
+        talon.actions.user.cursorless_everywhere_get_editor_state();
+      ide.updateTextEditors(editorState);
```

Comment:
prob worth a comment about why we need to do this. Ideally we wouldn't have to use this hack, but it is the right place to start